reports/3_generate_report_emails.py

```diff
@@ -0,0 +1,144 @@
+from calitp.tables import tbl
+from siuba import *
+import sys
+from postmarker.core import PostmarkClient
+import os
+import pandas as pd
+
+# +
+SERVER_TOKEN=os.environ["POSTMARK_SERVER_TOKEN"]
+
+if len(sys.argv) < 2:
+    raise Exception (
+        "2 arguments required: report publish date YYYY-MM-DD and Month"
+    )
+else:
+    REPORT_PUBLISH_DATE = sys.argv[1:]
+
+#REPORT_PUBLISH_DATE = "2021-07-01"
+#REPORT_PUBLISH_MONTH = "October"
+REPORT_PUBLISH_DATE = str(sys.argv[1])
+REPORT_PUBLISH_MONTH = str(sys.argv[2])
+print("Report Publish Date is:",REPORT_PUBLISH_DATE)
+print("Report Publish Month is:",REPORT_PUBLISH_MONTH)
+split_date = REPORT_PUBLISH_DATE.split("-")    
+PUBLISH_DATE_YEAR = split_date[0]
+PUBLISH_DATE_MONTH= split_date[1]
+
+REPORT_LINK_BASE = f'https://reports.calitp.org/gtfs_schedule/{PUBLISH_DATE_YEAR}/{PUBLISH_DATE_MONTH}'
+DEV_LINK_BASE = f'https://development-build--cal-itp-reports.netlify.app/gtfs_schedule/{PUBLISH_DATE_YEAR}/{PUBLISH_DATE_MONTH}'    
+
+
+# +
+df_crm_emails = pd.read_csv(
+    "https://docs.google.com/spreadsheets"
+    "/d/1AHFa5SKcEn7im374mPwULFCj7VlFchQr/export?gid=289636985&format=csv",
+    skiprows=1,
+)
+
+df_crm_emails.columns = df_crm_emails.columns.str.lower()
+
+# +
+tbl.gtfs_schedule.feed_info() >> count(missing_email = _.feed_contact_email.isna())
+
+tbl_report_emails = (
+    tbl.views.reports_gtfs_schedule_index()
+    >> left_join(
+        _,
+        tbl.gtfs_schedule.feed_info()
+        >> select(_.startswith("calitp"), _.feed_contact_email),
+        ["calitp_itp_id", "calitp_url_number"]
+    )
+    # NOTE: reports currently only use url number = 0, so we remove any others --
+    >> filter(_.publish_date == REPORT_PUBLISH_DATE,  _.calitp_url_number == 0)
+    >> select(_.startswith("calitp"), _.agency_name, _.use_for_report, _.feed_contact_email)
+)
+
+report_emails = tbl_report_emails >> collect()
+# -
+
+report_crm_emails = (
+    report_emails
+    >> full_join(
+        _,
+        df_crm_emails
+        >> filter(_.itp_id.notna())
+        >> mutate(itp_id=_.itp_id.astype(int)),
+        {"calitp_itp_id": "itp_id"},
+    )
+)
+
+report_crm_emails_inner = (
+    report_crm_emails
+    >> filter(_.calitp_itp_id.notna(), _.itp_id.notna())
+    >> select(-_.itp_id)
+    >> mutate(calitp_itp_id=_.calitp_itp_id.astype(int))
+)
+
+
+final_all_emails = (
+    report_crm_emails_inner
+    >> transmute(_.calitp_itp_id, _.calitp_url_number, email = _["main email"], origin = "CRM")
+    >> filter(_.email.notna())
+    >> mutate(
+        calitp_itp_id = _.calitp_itp_id.astype(int),
+        calitp_url_number = _.calitp_url_number.astype(int),
+        report_url = REPORT_LINK_BASE + "/" + _.calitp_itp_id.astype(str) + "/",
+        dev_url = DEV_LINK_BASE + "/" + _.calitp_itp_id.astype(str) + "/",
+    )
+)
```

Comment:
I'm a little confused about what all of this is doing and why it is needed. It seems like at a really basic level, all that is needed is to send an email to someone at a specific address and populate the html body with a link to an address with the appropriate itp_id. Both the email address and the itp_id are contained within the google sheet, so it seems like just the data from the google sheet could be used.